From 6e1a81242a473c6c95fcaa1cf7413701b222a5a6 Mon Sep 17 00:00:00 2001
From: jjacky <i.am.jack.mail@gmail.com>
Date: Wed, 24 Apr 2013 14:17:42 +0200
Subject: [PATCH] tooltip: Fix possible wrong placement

When showing a tooltip on the edge of a monitor, the tooltip could be wrongly
placed and be shown going from one monitor to the next.

This happened because the current_window wasn't set visible, and when it wasn't
the returned allocated size would be 1, hence wrong calculations.

(See b495ce5446a0 for why the allocated size is 1 when not visible.)

Signed-off-by: jjacky <i.am.jack.mail@gmail.com>
---
 gtk/gtktooltip.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gtk/gtktooltip.c b/gtk/gtktooltip.c
index aba9ffb..424b29d 100644
--- a/gtk/gtktooltip.c
+++ b/gtk/gtktooltip.c
@@ -1071,6 +1071,7 @@ gtk_tooltip_position (GtkTooltip *tooltip,
 #define MAX_DISTANCE 32
 
   gtk_widget_realize (GTK_WIDGET (tooltip->current_window));
+  gtk_widget_set_visible (GTK_WIDGET (tooltip->current_window), TRUE);
 
   tooltip->tooltip_widget = new_tooltip_widget;
 
-- 
1.8.2.1

